plugins/rest/aws.go

```diff
@@ -267,6 +274,13 @@ func (cs *awsMetadataCredentialService) refreshFromService(ctx context.Context)
 		return errors.New("unable to construct metadata HTTP request: " + err.Error())
 	}
 
+	// if using the AWS_CONTAINER_CREDENTIALS_FULL_URI variable, we need to associate the token
+	// to the request
+	if _, useFullPath := os.LookupEnv(ecsFullPathEnvVar); useFullPath {
+		token := os.Getenv(ecsAuthorizationTokenEnvVar)
+		req.Header.Set("Authorization", token)
```

Comment:
Should we return an error if the token is not set?

Reply:
Done!